SemanticsTests.fs

```diff
@@ -178,91 +142,48 @@ module Normalisation =
                          AVar (Reg "x"),
                          IInt 3L),
                      IVar (Reg "i"))),
-               Int (IInt 3L))
+               Some (Int (IInt 3L)))
               (Int
                 (IIdx
                     (Int (),
                      Some 320,
                      AVar (Reg "y"),
                      (IVar (Reg "j")))),
-               Int (IInt 4L)) ]
+               Some (Int (IInt 4L))) ]
 
     [<Test>]
     let ``microcode normalisation of CAS(x[3], d[6], 2) is correct`` () =
         let xel = Int ()
         let xlen = Some 10
-        let xname = Reg "x"
+        let xname = "x"
         let xar = Array (xel, xlen, xname)
-        let x3 = IIdx (xel, xlen, AVar xname, IInt 3L)
-        let x3upd v = aupd xel xlen (AVar xname) (IInt 3L) v
+        let x3 = IIdx (xel, xlen, AVar (Reg xname), IInt 3L)
+        let x3upd v = aupd xel xlen (AVar (Reg xname)) (IInt 3L) v
 
         let del = Int ()
         let dlen = Some 10
-        let dname = Reg "d"
+        let dname = "d"
         let dar = Array (del, dlen, dname)
-        let d6 = IIdx (del, dlen, AVar dname, IInt 6L)
-        let d6upd v = aupd del dlen (AVar dname) (IInt 6L) v
+        let d6 = IIdx (del, dlen, AVar (Reg dname), IInt 6L)
+        let d6upd v = aupd del dlen (AVar (Reg dname)) (IInt 6L) v
 
         checkMicrocode
             // TODO(CaptainHayashi): order shouldn't matter in branches.
             [ Branch
                 (iEq x3 d6,
-                 [ Assign (dar, d6upd (Int d6))
-                   Assign (xar, x3upd (Int (IInt 2L))) ],
-                 [ Assign (dar, d6upd (Int x3))
-                   Assign (xar, x3upd (Int x3)) ])
+                 [ dar *<- d6upd (Int d6)
+                   xar *<- x3upd (Int (IInt 2L)) ],
+                 [ dar *<- d6upd (Int x3)
+                   xar *<- x3upd (Int x3) ])
             ]
             [ Branch
                 (iEq x3 d6,
-                 [ Assign (Int x3, Int (IInt 2L))
-                   Assign (Int d6, Int d6) ],
-                 [ Assign (Int x3, Int x3)
-                   Assign (Int d6, Int x3) ])
+                 [ Int x3 *<- Int (IInt 2L)
+                   Int d6 *<- Int d6 ],
+                 [ Int x3 *<- Int x3
+                   Int d6 *<- Int x3 ])
             ]
 
-module Frames =
```

Comment:
Old framer is gone, so these tests are too.  The framer is exercised in the higher-level Boolean construction tests and in the `MakeFrame` module above.